src/Button/index.js

```diff
@@ -0,0 +1,271 @@
+      textSize: OIOResponsiveObjectPropType,
+      textTransform: OIOResponsiveObjectPropType,
+      textWeight: OIOResponsiveObjectPropType,
+      type: PropTypes.oneOf(['button', 'clear', 'submit']),
+      width: OIOResponsiveObjectPropType
+   }
+
+   static defaultProps = {
+      borderRadius: r`4px`,
+      children: undefined,
+      mode: 'normal',
+      onClick: () => {},
+      outline: false,
+      rounded: false,
+      size: r`md`,
+      style: {},
+      tagName: 'button',
+      textColor: r`#fff`,
+      textSize: undefined,
+      textTransform: r`none`,
+      textWeight: r`medium`,
+      type: 'button',
+      width: r`auto`
+   }
+
+   render() {
+      const {
+         children, className, mode, name, tagName, type,
+         textColor, textSize, textTransform, textWeight, onClick,
+         backgroundColor, border, borderRadius, color, fontFamily,
+         height, padding, minWidth, width,
+         hoverBackgroundColor, hoverBorder
+      } = this.props
+
+      /* eslint-disable object-property-newline */
+      const responsiveStyles = generateResponsiveStyles({
+         backgroundColor, border, borderRadius, color, fontFamily,
+         height, padding, minWidth, width
+      })
+
+      const hoverResponsiveStyles = generateResponsiveStyles({
+         backgroundColor: hoverBackgroundColor,
+         border: hoverBorder
+      })
+      /* eslint-enable */
+
+      const textStyle = {}
+      const buttonStyle = {
+         alignItems: 'center',
+         cursor: 'pointer',
+         display: 'inline-flex',
+         justifyContent: 'center',
+         outline: 'none',
+         position: 'relative',
+         transition: '200ms',
+         ...this.props.style,
+         ...responsiveStyles
+      }
+
+      // ====================================================
+      // Element
+      // ====================================================
+
+      // Abilty to use custom tag (div, button, span)
+      const ButtonElement = tagName
+      const conditionalButtonProps = {}
+
+      // If Button is used <button>, add type attribute
+      if (tagName === 'button') {
+         conditionalButtonProps.type = type
+      }
+
+      // ====================================================
+      // Mode
+      // ====================================================
+
+      if (mode === 'disabled') {
+         buttonStyle.opacity = '0.3'
+         buttonStyle.cursor = 'default'
+         conditionalButtonProps.disabled = true
+      } else if (mode === 'loading') {
+         textStyle.opacity = 0
+         buttonStyle.cursor = 'default'
+         conditionalButtonProps.disabled = true
+      } else if (mode === 'pulsing') {
+         buttonStyle.animation = `${pulsingAnimation} 2000ms infinite linear`
+      }
+
+      // Enable hover and active styles for normal and pulsing button modes
+      // No hover/active styles for modes: loading or disabled
+      if (mode === 'normal' || mode === 'pulsing') {
```

Comment:
No change required, this is more just a tip that you may already know and useful in some cases if the OR statement gets hairy. 

`if (['normal', 'pulsing'].includes(mode)) {`